### DISCUSSION TITLE
`wrapmargin` but for soft-wrapping

### DISCUSSION THREAD

**Discussion by LunarLambda** [neovim/neovim#27237] [Ideas]
I find text that wraps a few cells before the edge of the window to be easier to parse visually, on top of being subjectively more aesthetically pleasing.

I actually thought the `wrapmargin` option did exactly that, but rather than being related to `wrap`, it's actually an alternative to `textwidth`.

**Comment by poperigby** [comment]
Yes, this is a much needed feature. Being able to set the width at which text wraps in a non-destructive way would be very nice.

**Comment by HunterretnuH** [comment]
It would also be really useful when using plugins like nvim-scrollbar or nvim-scrollview.

**Comment by smackesey** [comment]
Been waiting for soft-wrapping independent of vim window size for 10 years.

**Reply by justinmk** [reply]
Tracked in https://github.com/neovim/neovim/issues/4386
